### DISCUSSION TITLE
Compute error between functions from different meshes?

### DISCUSSION THREAD

**Discussion by gatling-nrl** [kinnala/scikit-fem#916] [Q&A]
I want to compute the integral of the squared difference between one P0 function on a fine mesh and another P0 function on a different (usually coarser) mesh over the same region of space. The meshes are independent of each other (i.e., they don't share some subset of vertices).

I could brute force what I want to do using `.probes` to query the functions at a  common set of coordinates (i.e. the quadrature points from the fine mesh). But is there a better way?

**Comment by kinnala** [comment]
Unfortunately there is no way currently implemented.

Technically one would need a subroutine which combines two meshes by creating a new "supermesh". Each triangle in the "supermesh" has its edges so that they do not cross any of the edges in the original meshes. Each triangle also needs to store which two original triangles it is part of.

There is a Forran library (libsupermesh) for creating such combined meshes. Firedrake implements bindings to this library using Cython.

**Reply by gatling-nrl** [reply]
Is using `.probes` to query the coarse function at the `.global_coordinates()` of the finer function and then summing the square of the difference a good "next best thing?"

For example:
```python
interpolate_to_fine_p0 = coarse_p0.probes(fine_p0.global_coordinates().reshape(2, -1)
coarse_on_fine = interpolate_to_fine_p0 @ coarse_func
fine_on_fine = fine_p0.interpolate(fine_func)
error = ((coarse_on_fine - fine_on_fine)**2).sum()
```

Am I'm missing a scale factor somewhere? In the "\int (f -g)^2 dx" I think I'm not accounting for the dx in the way I described.

**Reply by kinnala** [reply]
I would probably use Basis.interpolator() inside Functional form to calculate the difference.
